kernel/kernel-auth-service/src/main/java/io/mosip/kernel/auth/service/impl/ProxyAuthServiceImpl.java

```diff
@@ -0,0 +1,648 @@
+			authNResponseDto.setStatus(authNResponseDto.getStatus());
+			authNResponseDto.setMessage(authNResponseDto.getMessage());
+		} else {
+			throw new AuthManagerException(String.valueOf(HttpStatus.UNAUTHORIZED.value()), "Invalid User Id type");
+		}
+		return authNResponseDto;
+	}
+
+	/**
+	 * Method used for Authenticating User based with username and OTP
+	 * 
+	 * @param userOtp is of type {@link UserOtp}
+	 * 
+	 * @return authNResponseDto is of type {@link AuthNResponseDto}
+	 * 
+	 * @throws Exception exception
+	 * 
+	 */
+
+	@Override
+	public AuthNResponseDto authenticateUserWithOtp(UserOtp userOtp) throws Exception {
+		AuthNResponseDto authNResponseDto = new AuthNResponseDto();
+		MosipUserTokenDto mosipToken = null;
+		MosipUserDto mosipUser = null;
+		String realm = realmId;
+		if (userOtp.getAppId().equalsIgnoreCase(AuthConstant.PRE_REGISTRATION)) {
+			realm = userOtp.getAppId();
+		}
+		mosipUser = new MosipUserDto();
+		mosipUser.setUserId(userOtp.getUserId());
+		if (mosipUser == null && AuthConstant.IDA.toLowerCase().equals(userOtp.getAppId().toLowerCase())) {
+			mosipUser = uinService.getDetailsForValidateOtp(userOtp.getUserId());
+		}
+		if (mosipUser != null) {
+			mosipToken = oTPService.validateOTP(mosipUser, userOtp.getOtp(), userOtp.getAppId());
+		} else {
+			throw new AuthManagerException(AuthErrorCode.USER_VALIDATION_ERROR.getErrorCode(),
+					AuthErrorCode.USER_VALIDATION_ERROR.getErrorMessage());
+		}
+		if (mosipToken != null && mosipToken.getMosipUserDto() != null) {
+			authNResponseDto.setMessage(mosipToken.getMessage());
+			authNResponseDto.setStatus(mosipToken.getStatus());
+			authNResponseDto.setToken(mosipToken.getToken());
+			authNResponseDto.setExpiryTime(mosipToken.getExpTime());
+			authNResponseDto.setRefreshToken(mosipToken.getRefreshToken());
+			authNResponseDto.setUserId(mosipToken.getMosipUserDto().getUserId());
+		} else {
+			authNResponseDto.setMessage(mosipToken.getMessage());
+			authNResponseDto.setStatus(mosipToken.getStatus());
+		}
+		return authNResponseDto;
+	}
+
+	/**
+	 * Method used for Authenticating User based with secretkey and password
+	 * 
+	 * @param clientSecret is of type {@link ClientSecret}
+	 * 
+	 * @return authNResponseDto is of type {@link AuthNResponseDto}
+	 * 
+	 * @throws Exception exception
+	 * 
+	 */
+
+	@Override
+	public AuthNResponseDto authenticateWithSecretKey(ClientSecret clientSecret) throws Exception {
+		return proxyTokenForLocalEnv(clientSecret.getClientId(), SUCCESS,
+				CLIENTID_AND_TOKEN_COMBINATION_HAD_BEEN_VALIDATED_SUCCESSFULLY);
+	}
+
+	private AuthNResponseDto proxyTokenForLocalEnv(String subject, String status, String message) {
+		long exp = System.currentTimeMillis() + localExp;
+		String token = proxyTokenGenarator.getProxyToken(subject, exp);
+		AuthNResponseDto authNResponseDto = new AuthNResponseDto();
+		authNResponseDto.setToken(token);
+		authNResponseDto.setRefreshToken(null);
```

Comment:
Why do we set this to be empty?

Reply:
Refresh token is not currently used by any module except kernel auth service which calls keycloak with access and refresh token no other module is following this paradigm.